solr/core/src/java/org/apache/solr/util/stats/MetricUtils.java

```diff
@@ -385,10 +406,10 @@ static double nsToMs(boolean convert, double value) {
   }
 
   // some snapshots represent time in ns, other snapshots represent raw values (eg. chunk size)
-  static void addSnapshot(Map<String, Object> response, Snapshot snapshot, PropertyFilter propertyFilter, boolean ms) {
+  static void addSnapshot(MapWriter.EntryWriter ew, Snapshot snapshot, PropertyFilter propertyFilter, boolean ms) {
     BiConsumer<String, Object> filter = (k, v) -> {
       if (propertyFilter.accept(k)) {
```

Comment:
I see this pattern repeated in many places. I think a helper method can be added to avoid code duplication

Reply:
Which pattern?